test/acceptance/tools/mongo_utils.py

```diff
@@ -0,0 +1,212 @@
+# -*- coding: utf-8 -*-
+"""
+(c) Copyright 2014 Telefonica, I+D. Printed in Spain (Europe). All Rights
+Reserved.
+
+The copyright to the software program(s) is property of Telefonica I+D.
+The program(s) may be used and or copied only with the express written
+consent of Telefonica I+D or in accordance with the terms and conditions
+stipulated in the agreement/contract under which the program(s) have
+been supplied.
+"""
+import time
+
+__author__ = 'Iván Arias León (ivan.ariasleon@telefonica.com)'
+
+
+import pymongo
+
+# constants
+EMPTY          = u''
+HOST           = u'host'
+HOST_DEFAULT   = u'localhost'
+PORT           = u'port'
+PORT_DEFAULT   = u'27017'
+USER           = u'user'
+PASSWORD       = u'password'
+DATABASE       = u'database'
+COLLECTION     = u'collection'
+BUILD_INFO     = u'buildInfo'
+VERSION        = u'version'
+VERIFY_VERSION = u'verify_version'
+RETRIES        = u'retries'
+RETRY_DELAY    = u'retry_delay'
+
+
+class Mongo:
+    """
+    Mongo Management Class
+    """
+
+    def __init__(self,  **kwarg):
+        """
+        constructor class
+        :param host:  mongo hostname
+        :param port: mongo port
+        :param user:  mongo username
+        :param password: password associated to mongo username
+        :param version: mongo version
+        :param verify_version: determine if the mongo version is verified ( true |false )
+        :param database: mongo current database
+        :param collection: mongo current collection
+        :param retries: number of retries when find documents
+        :param retry_delay: delay en each retry when find documents
+        """
+        self.host            = kwarg.get(HOST, HOST_DEFAULT)
+        self.port            = kwarg.get(PORT, PORT_DEFAULT)
+        self.user            = kwarg.get(USER, EMPTY)
+        self.password        = kwarg.get(PASSWORD, EMPTY)
+        self.version         = kwarg.get(VERSION, EMPTY)
+        self.verify_version  = kwarg.get(VERIFY_VERSION, EMPTY)
+        self.database_name   = kwarg.get(DATABASE, EMPTY)
+        self.collection_name = kwarg.get(COLLECTION, EMPTY)
+        self.retries         = kwarg.get(RETRIES, 1)
+        self.retry_delay     = kwarg.get(RETRY_DELAY, 1)
+
+    def connect(self, database=EMPTY):
+        """
+        connect to mongo
+        Standard URI format: mongodb://[dbuser:dbpassword@]host:port/dbname
+        """
+        if database != EMPTY:
+            self.database_name = database
+        self.mongo_uri = "mongodb://%s:%s/%s" % (self.host, self.port, self.database_name)
+        try:
+            self.client = pymongo.MongoClient(self.mongo_uri)
+            self.current_database = self.client.get_default_database()
+            if self.collection_name != u'':
+                self.current_collection = self.current_database[self.collection_name]
```

Comment:
This instance attribute "current_collection" is not set in the init, if the method "get_current_collection" is called before the connect, an exception is going to be raised because the variable doesnt exist